hoomd/data/typeparam.py

```diff
@@ -110,33 +69,93 @@ def __getattr__(self, attr):
             raise AttributeError("'{}' object has no attribute "
                                  "'{}'".format(type(self), attr))
 
-    def __getitem__(self, key):
-        """Access parameters by key."""
-        return self.param_dict[key]
-
     def __setitem__(self, key, value):
-        """Set parameters by key."""
+        """Set parameters for a given type (or type pair).
+
+        .. rubric:: Examples:
+
+        Index types by name:
+
+        .. skip: next if(not hoomd.version.md_built)
+        .. code-block:: python
+
+            langevin.gamma['A'] = 2.0
+
+        Set parameters for multiple types:
+
+        .. skip: next if(not hoomd.version.md_built)
+        .. code-block:: python
+
+            langevin.gamma[['B', 'C']] = 3.0
+
+        Set type pair parameters with a tuple of names:
+
+        .. skip: next if(not hoomd.version.md_built)
+        .. code-block:: python
+
+            lj.params[('A', 'A')] = dict(epsilon=1.5, sigma=2.0)
+
+        Set parameters for multiple pairs (e.g. ('A', 'B') and ('A', 'C')):
+
+        .. skip: next if(not hoomd.version.md_built)
+        .. code-block:: python
+
+            lj.params[('A', ['B', 'C'])] = dict(epsilon=0, sigma=0)
+
+        Set parameters for multiple pairs (e.g. ('B', 'B'), ('B', 'C'), ('C',
+        'B'), and ('C', 'C')):
+
+        .. skip: next if(not hoomd.version.md_built)
+        .. code-block:: python
+
+            lj.params[(['B', 'C'], ['B', 'C'])] = dict(epsilon=1, sigma=1)
```

Comment:
I get the reason to include `('C', 'B')`, but that won't be set actually. Just wanted to mention this.

Reply:
Yes, that note is preserved on the next line. I listed both pairs here for completeness.